zowex/src/main.rs

```diff
@@ -279,30 +290,48 @@ fn run_daemon_command(args: &mut Vec<String>) -> io::Result<()> {
         _resp.append(&mut stdin);
     }
 
-    // form our host, port, and connection strings
-    let daemon_host = "127.0.0.1".to_owned();
-    let port_string = get_port_string();
+    let mut tries = 0;
+    loop {
+        let socket_string = get_socket_string();
```

Comment:
This could be moved outside of the loop, if we change the `establish_connection` method to take a pointer so that Rust doesn't think the variable passed into it has moved.
L296: `let mut stream = establish_connection(&socket_string)?;`
L326: `fn establish_connection(daemon_socket: &str) -> io::Result<DaemonClient> {`

Reply:
Done